femwell/maxwell/waveguide.py

```diff
@@ -69,6 +69,45 @@ def n_eff(self):
         """Effective refractive index of the mode"""
         return self.k / self.k0
 
+    @cached_property
+    def poynting(self):
+        """Poynting vector of the mode"""
+
+        # Extraction of the fields
+        (Ex, Ey), Ez = self.basis.interpolate(self.E)
+        (Hx, Hy), Hz = self.basis.interpolate(self.H)
+
+        # New basis will be the discontinuous variant used by the solved Ez-field
+        (Et, Et_basis), (En, En_basis) = self.basis.split(self.E)
+        poynting_basis = self.basis.with_element(ElementDG(ElementTriP1()))
+
+        # Calculation of the Poynting vector
+        Px = Ey * Hz - Ez * Hy
+        Py = Ez * Hx - Ex * Hz
+        Pz = Ex * Hy - Ey * Hx
+
+        # Projection of the Poynting vector on the new basis
+        Px_proj = poynting_basis.project(Px, dtype=np.complex64)
+        Py_proj = poynting_basis.project(Py, dtype=np.complex64)
+        Pz_proj = poynting_basis.project(Pz, dtype=np.complex64)
+
+        return poynting_basis, np.array([Px_proj, Py_proj, Pz_proj])
+
+    @cached_property
+    def Px(self):
+        basis, _P = self.poynting
+        return basis, _P[0]
+
+    @cached_property
+    def Py(self):
+        basis, _P = self.poynting
+        return basis, _P[1]
+
+    @cached_property
+    def Pz(self):
+        basis, _P = self.poynting
+        return basis, _P[2]
```

Comment:
As soon as we use the vector-basis, we don't need these three functions anymore as the result can be treated like E/H